### PR TITLE
Adds badge route to dream-web.

### PR DESCRIPTION
**Pull request by novemberkilo**

### CODE REVIEW

File: dream-web/controller/badges.ml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+type schema = {
+  schema_version : int; [@key "schemaVersion"]  (** always: 1 *)
```
Comment (maiste) [inline dream-web/controller/badges.ml:2]: Couldn't the `[@default 1]` be added to enforce the condition?

File: dream-web/controller/badges.ml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+let schema_of_status =
+  let v = v ~label:"ocaml-ci" in
+  function
+  | `Not_started -> v ~message:"unknown" ~color:"inactive" ()
+  | `Pending -> v ~message:"building" ~color:"yellow" ()
+  | `Failed -> v ~message:"failing" ~color:"critical" ()
+  | `Passed -> v ~message:"passing" ~color:"success" ()
+
+let ( let* ) = Lwt.Infix.( >>= )
```
Comment (maiste) [inline dream-web/controller/badges.ml:47]:
```suggestion
let ( let* ) = Lwt.Syntax.( let* )
```

Comment (maiste) [reply dream-web/controller/badges.ml:47]: And why use `(let*)` as it doesn't seem to be used anywhere? :wink:

**Review by maiste — COMMENTED** [review]
Small typos but otherwise it LGTM :+1:

File: dream-web/controller/badges.ml (hunk 1 above)
Comment (tmcgilchrist) [reply dream-web/controller/badges.ml:2]: Should check if the `@default value` can be used to enforce a stricter typing here.

**Pull request merged**